packages/salesforcedx-vscode-apex/src/index.ts

```diff
@@ -72,56 +72,64 @@ export async function activate(context: vscode.ExtensionContext) {
     coreTelemetryService.isTelemetryEnabled()
   );
 
-  const langClientHRStart = process.hrtime();
-  languageClient = await languageServer.createLanguageServer(context);
-  LanguageClientUtils.setClientInstance(languageClient);
-  const handle = languageClient.start();
-  context.subscriptions.push(handle);
+  // Initialize Apex language server
+  try {
+    const langClientHRStart = process.hrtime();
+    languageClient = await languageServer.createLanguageServer(context);
+    languageClientUtils.setClientInstance(languageClient);
+    const handle = languageClient.start();
+    languageClientUtils.setStatus(ClientStatus.Indexing, '');
+    context.subscriptions.push(handle);
 
-  languageClient
-    .onReady()
-    .then(async () => {
-      if (languageClient) {
-        languageClient.onNotification('indexer/done', async () => {
-          LanguageClientUtils.indexing = false;
+    languageClient
+      .onReady()
+      .then(async () => {
+        if (languageClient) {
+          languageClient.onNotification('indexer/done', async () => {
+            // Refresh SObject definitions if there aren't any faux classes
+            const sobjectRefreshStartup: boolean = vscode.workspace
+              .getConfiguration(SFDX_APEX_CONFIGURATION_NAME)
+              .get<boolean>(ENABLE_SOBJECT_REFRESH_ON_STARTUP, false);
+            if (sobjectRefreshStartup) {
+              initSObjectDefinitions(getRootWorkspacePath()).catch(e =>
+                telemetryService.sendErrorEvent(e.message, e.stack)
+              );
+            }
 
-          // Refresh SObject definitions if there aren't any faux classes
-          const sobjectRefreshStartup: boolean = vscode.workspace
-            .getConfiguration(SFDX_APEX_CONFIGURATION_NAME)
-            .get<boolean>(ENABLE_SOBJECT_REFRESH_ON_STARTUP, false);
-          if (sobjectRefreshStartup) {
-            initSObjectDefinitions(getRootWorkspacePath()).catch(e =>
-              telemetryService.sendErrorEvent(e.message, e.stack)
-            );
-          }
-
-          await testOutlineProvider.refresh();
-        });
-      }
-      LanguageClientUtils.languageClientReady = true;
-      telemetryService.sendApexLSPActivationEvent(langClientHRStart);
-    })
-    .catch(err => {
-      // Handled by clients
-      telemetryService.sendApexLSPError(err);
-    });
+            await testOutlineProvider.refresh();
+          });
+        }
+        languageClientUtils.setStatus(ClientStatus.Ready, '');
+        telemetryService.sendApexLSPActivationEvent(langClientHRStart);
+      })
+      .catch(err => {
+        // Handled by clients
+        telemetryService.sendApexLSPError(err);
+        languageClientUtils.setStatus(ClientStatus.Error, err);
```

Comment:
I haven't test this enough but this will only happen when an internal Apex language server happens. Should we add a generic message instead of directly pushing the error ?

Reply:
What if we set a generic message and followed it up with the error?

Reply:
I think we should add a generic message here if it is going to be visible to customer since the actual error might not make sense for the customer. We have telemetry of the actual error that should help us debug the issue.